### PR TITLE
Add text report.

### PR DESCRIPTION
**Pull request by nlamirault**
A simple code coverage which display metrics.

Configure undercover to use this report style : 

``` lisp
(undercover "*.el"
                  (:report-type 'text))
```

``` bash
$ cask exec ert-runner
[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el
[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el
[dionysos-vlc] Start /home/nlamirault/Perso/dionysos/test/resources/Roulement_tambour-01.mp3 Next nil
[dionysos-process] Create dionysos vlc (--intf rc /home/nlamirault/Perso/dionysos/test/resources/Roulement_tambour-01.mp3 vlc://quit) nil
[dionysos-process] Status : nil
[dionysos-vlc] Stop
[dionysos-process] Process dionysos event:killed
 status:9
[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el
[dionysos-process] Create dionysos mpg123 (/home/nlamirault/Perso/dionysos/test/resources/Roulement_tambour-01.mp3) (lambda nil (message ok))
[dionysos-process] Status : nil
[dionysos-process] Process dionysos event:killed
 status:9

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el
[dionysos-process] Create dionysos mpg123 (/home/nlamirault/Perso/dionysos/test/resources/France.mp3) (lambda nil (message ok))
[dionysos-process] Status : nil
[dionysos-process] Create dionysos mpg123 (/home/nlamirault/Perso/dionysos/test/resources/France.mp3) (lambda nil (message ok))
[dionysos-process] Status : run
Process multiple : (dionysos)
[dionysos-process] Process dionysos event:killed
 status:9

[dionysos] Cleanup path
[dionysos] Load library from /home/nlamirault/Perso/dionysos/dionysos.el
dionysos 0.7.0
....................

Ran 20 tests in 0.198 seconds

== Code coverage text report ==

dionysos-mpd-mode : Percent 5% [Relevant: 73 Covered: 4 Missed: 69]
dionysos-fs-mode : Percent 6% [Relevant: 66 Covered: 4 Missed: 62]
dionysos-mode : Percent 0% [Relevant: 17 Covered: 0 Missed: 17]
dionysos-volume : Percent 0% [Relevant: 14 Covered: 0 Missed: 14]
dionysos-backend-mpd : Percent 8% [Relevant: 25 Covered: 2 Missed: 23]
dionysos-backend-mplayer : Percent 0% [Relevant: 4 Covered: 0 Missed: 4]
dionysos-backend-vlc : Percent 69% [Relevant: 13 Covered: 9 Missed: 4]
dionysos-backend : Percent 100% [Relevant: 32 Covered: 32 Missed: 0]
dionysos-ui : Percent 0% [Relevant: 6 Covered: 0 Missed: 6]
dionysos-process : Percent 78% [Relevant: 28 Covered: 22 Missed: 6]
dionysos-notify : Percent 100% [Relevant: 3 Covered: 3 Missed: 0]
dionysos-io : Percent 5% [Relevant: 17 Covered: 1 Missed: 16]
dionysos-version : Percent 25% [Relevant: 8 Covered: 2 Missed: 6]

```

### CODE REVIEW

**Comment by sviridov** [comment]
Hi @nlamirault, thanks for the PR!

I will review it in one or two days.

File: undercover.el [hunk 1]
```diff
@@ -386,14 +395,48 @@ Values of that hash are number of covers."
...
+;; Text report
+;; ------------
+
+
+(defun undercover--create-text-report ()
+  "Create test coverage report for text display."
+  (undercover--collect-files-coverage undercover--files)
+  (let ((report (make-hash-table :test 'equal)))
+    (maphash (lambda (key val)
+               ;; (message "> %s" key)
```
Comment (sviridov) [inline undercover.el]: Please remove commented code

File: undercover.el [hunk 2]
```diff
@@ -386,14 +395,48 @@ Values of that hash are number of covers."
...
+               (let ((nb 0)
+                     (covered 0))
+                 (maphash (lambda (k v)
+                            ;;(message "--> %s %s" k v)
```
Comment (sviridov) [inline undercover.el]: Please remove commented code

File: undercover.el [hunk 3]
```diff
@@ -232,7 +236,12 @@ Values of that hash are number of covers."
 
 (defun undercover--determine-report-type ()
   "Automatic report-type determination."
-  (and (undercover--under-ci-p) 'coveralls))
+  (cond ((eq 'coveralls undercover--report-type)
+         'coveralls)
+        ((eq 'text undercover--report-type)
+         'text)
+        (t 'coveralls)))
+;;  (and (undercover--under-ci-p) 'coveralls))
```
Comment (sviridov) [inline undercover.el]: We cannot ignore `undercover--under-ci-p` check. That is my proposal:

```emacs-lisp
(defvar undercover--report-type 'coveralls
  "Type of code coverage report.")

(defun undercover--determine-report-type ()
   "Automatic report-type determination."
  (cond ((and (eq 'coveralls undercover--report-type) (undercover--under-ci-p))
         'coveralls)
        ((eq 'text undercover--report-type)
         'text)))
```

File: undercover.el [hunk 4]
```diff
@@ -386,14 +395,48 @@ Values of that hash are number of covers."
...
+;; ------------
+;; Text report
+;; ------------
+
+
+(defun undercover--create-text-report ()
+  "Create test coverage report for text display."
+  (undercover--collect-files-coverage undercover--files)
+  (let ((report (make-hash-table :test 'equal)))
+    (maphash (lambda (key val)
```
Comment (sviridov) [inline undercover.el]: Please extract both lambdas to separated functions and give arguments meaningful name. To be honest it's hard to understand what `k, v, nb` means.

**Comment by sviridov** [comment]
Hi @nlamirault, sorry it took so long!

I added few comments to code, hope you will not be against it.

P.S. Please add section about this reporter to README.md.

**Comment by CyberShadow** [comment]
Revived in #51.

**Pull request closed without merging**